js/src/lib/components/DropdownMenu/DropdownMenu.svelte

```diff
@@ -12,22 +12,28 @@
 	// the browser won't be able to properly compute x and width
 	let alignement: Alignement = forceAlignement ?? "left";
 	let element: HTMLElement | undefined;
+	let docWidth: number | undefined;
 
 	onMount(() => {
 		document.addEventListener("click", handleClickDocument);
-		window.addEventListener("blur", onClose);
+		screen?.orientation?.addEventListener("change", onClose);
 
+		docWidth = document.documentElement.clientWidth;
+		const bbox = element?.getBoundingClientRect();
+		const left = bbox.left ?? 0;
+		const width = bbox.width ?? 0;
 		if (!forceAlignement) {
-			const docWidth = document.documentElement.clientWidth;
-			const domRect = element?.getBoundingClientRect() || {};
-			const left = domRect["left"] ?? 0;
-			const width = domRect["width"] ?? 0;
 			alignement = left + width > docWidth ? "right" : "left";
 		}
+		if (alignement === "screen-center") {
+			element.style.transform = `translateX(${
+				docWidth / 2 - width / 2 - bbox.left
+			}px)`;
```

Comment:
would it be possible to use [inset-x-0 instead](https://tailwindcss.com/docs/top-right-bottom-left) ?

Reply:
Good point. Unfortunately, the menu is enclosed under multiple levels of parents with a `relative` position.  `inset-x-0` will center inside the closest relative element, not the whole screen. It can work if the menu has a `fixed` position but it lead to other issues to handle like the scrolling...